File: content/snippets/typescript/design_patterns/singleton.ts
class Config {
  private static shared: Config | null = null;

  private constructor(public env: string) {}

  static instance(): Config {
    if (!Config.shared) {
      Config.shared = new Config('dev');
    }
    return Config.shared;
  }
}

Config.instance().env = 'prod';
console.log(Config.instance().env);

File: content/snippets/go/design_patterns/singleton.go
package main

import (
	"fmt"
	"sync"
)

type Config struct {
	Env string
}

var (
	once   sync.Once
	shared *Config
)

func Instance() *Config {
	once.Do(func() {
		shared = &Config{Env: "dev"}
	})
	return shared
}

func main() {
	Instance().Env = "prod"
	fmt.Println(Instance().Env)
}
